tests/test_mig_shared_transferfunctions.py

```diff
@@ -0,0 +1,208 @@
+DUMMY_SETTINGS_PATH = os.path.join(TEST_OUTPUT_DIR, DUMMY_SETTINGS_DIR)
+
+
+def noop(*args, **kwargs):
+    if args:
+        return args[0]
+    return None
+
+
+class MigSharedTransferfunctions(MigTestCase):
+    """Wrap unit tests for the corresponding module"""
+
+    def test_transfers_basic_locking(self):
+        os.makedirs(os.path.join(DUMMY_HOME_PATH, DUMMY_USER))
+        cleanpath(DUMMY_HOME_DIR, self)
+        os.makedirs(os.path.join(DUMMY_SETTINGS_PATH, DUMMY_USER))
+        cleanpath(DUMMY_SETTINGS_DIR, self)
+        dummy_conf = FakeConfiguration(user_home=DUMMY_HOME_PATH,
+                                       user_settings=DUMMY_SETTINGS_PATH)
+        transfers_path = get_transfers_path(dummy_conf, DUMMY_USER)
+
+        # Lock shared twice should be fine
+        ro_lock = lock_data_transfers(transfers_path, exclusive=False)
+        ro_lock_again = lock_data_transfers(transfers_path, exclusive=False)
+        assert(ro_lock and ro_lock_again)
+
+        # Non-blocking repeated exclusive locking must fail
+        rw_lock_again = lock_data_transfers(
+            transfers_path, exclusive=True, blocking=False)
+        assert(not rw_lock_again)
+
+        # Unlock all to leave critical section and allow exclusive locking
+        unlock_data_transfers(ro_lock)
+        unlock_data_transfers(ro_lock_again)
+
+        # Take exclusive lock
+        rw_lock = lock_data_transfers(transfers_path, exclusive=True)
+        # Non-blocking repeated shared or exclusive locking must fail
+        ro_lock_again = lock_data_transfers(
+            transfers_path, exclusive=False, blocking=False)
+        rw_lock_again = lock_data_transfers(
+            transfers_path, exclusive=True, blocking=False)
+        assert(rw_lock and not ro_lock_again and not rw_lock_again)
+
+        unlock_data_transfers(rw_lock)
+
+    def test_create_and_delete_transfer(self):
+        os.makedirs(os.path.join(DUMMY_HOME_PATH, DUMMY_USER))
+        cleanpath(DUMMY_HOME_DIR, self)
+        os.makedirs(os.path.join(DUMMY_SETTINGS_PATH, DUMMY_USER))
+        cleanpath(DUMMY_SETTINGS_DIR, self)
+        dummy_conf = FakeConfiguration(user_home=DUMMY_HOME_PATH,
+                                       user_settings=DUMMY_SETTINGS_PATH)
+        (success, out) = create_data_transfer(dummy_conf, DUMMY_USER,
+                                              {'transfer_id': DUMMY_ID})
+        assert(success and DUMMY_ID in out)
+
+        (success, transfers) = load_data_transfers(dummy_conf, DUMMY_USER)
+
+        assert(success and transfers.get(DUMMY_ID, None))
+
+        (success, out) = delete_data_transfer(dummy_conf, DUMMY_USER, DUMMY_ID)
+        assert(success and out == DUMMY_ID)
+
+        (success, transfers) = load_data_transfers(dummy_conf, DUMMY_USER)
+
+        assert(success and transfers.get(DUMMY_ID, None) is None)
+
+    def test_transfers_shared_read_locking(self):
+        os.makedirs(os.path.join(DUMMY_HOME_PATH, DUMMY_USER))
```

Comment:
There is a lot of common setup in these methods, and with smaller and more localised test cases it would be potentially fruitful for the purposes of readability to reduce the amount of boilerplate in each test so one can clearly see the relevant test scenario, the activity and then assertions on a result. This is commonly called an "arrange, act, assert" structure of test cases. To get there in this case I'd suggest doing path setup once in a `def before_each()` block.